### ISSUE TITLE
DictField + EmbeddedDocumentFields

### ISSUE THREAD

**Issue by amcgregor** [hmarr/mongoengine#339]
I would like to be able to use an `EmbeddedDocumentField` instance (since you have to specify the type of embedded document) or just directly reference the `EmbeddedDocument` subclass as the base class used in a `DictField`.  E.g. I want to have a mapping of embedded documents.

No, I can't use an `EmbeddedDocument` subclass to store these key/value combinations as the keys are dynamic.

Currently there is an assert preventing the use if an instance of `EmbeddedDocumentField`, and `EmbeddedDocument` isn't a subclass of `BaseField`, it's a `BaseDocument` subclass.

How is the `basecls` argument to `DictField` _supposed_ to be used?

**Comment by rozza** [comment]
Can you provide some pseudo code of what you are proposing to do / trying to achieve?

**Comment by amcgregor** [comment]
``` python
class Credential(EmbeddedDocument):
    password = StringField()

class Identity(Document):
    credential = DictField(EmbeddedDocumentField(Credential))

owner = Identity()
owner.credential['admin'] = Credential(password="foo")
```

Terrible example, but it does illustrate what I'm trying to do.  I want the values associated with the keys to be managed like they are for list fields.

**Comment by amcgregor** [comment]
Ping.

**Comment by rozza** [comment]
Haven't got round to this yet 0.6 was just released today, this is scheduled for 0.7. patches welcome :-)

**Comment by amcgregor** [comment]
Moved to: https://github.com/MongoEngine/mongoengine/issues/168
